pkg/preference/request.go

```diff
@@ -0,0 +1,155 @@
+	Expires             *bool                                 `json:"expires,omitempty"`
+	ExternalReference   string                                `json:"external_reference,omitempty"`
+	Items               []PreferenceItemRequest               `json:"items,omitempty"`
+	Marketplace         string                                `json:"marketplace,omitempty"`
+	MarketplaceFee      float64                               `json:"marketplace_fee,omitempty"`
+	Metadata            map[string]interface{}                `json:"metadata,omitempty"`
+	NotificationUrl     string                                `json:"notification_url,omitempty"`
+	OperationType       string                                `json:"operation_type,omitempty"`
+	Payer               *PreferencePayerRequest               `json:"payer,omitempty"`
+	PaymentMethods      *PreferencePaymentMethodsRequest      `json:"payment_methods,omitempty"`
+	ProcessingModes     []string                              `json:"processing_modes,omitempty"`
+	Purpose             string                                `json:"purpose,omitempty"`
+	Shipments           *PreferenceShipmentsRequest           `json:"shipments,omitempty"`
+	StatementDescriptor string                                `json:"statement_descriptor,omitempty"`
+	Taxes               []PreferenceTaxRequest                `json:"taxes,omitempty"`
+	Tracks              []PreferenceTrackRequest              `json:"tracks,omitempty"`
+}
+
+// PreferenceBackUrlsRequest contains callback URLs.
+type PreferenceBackUrlsRequest struct {
+	Success string `json:"success,omitempty"`
+	Pending string `json:"pending,omitempty"`
+	Failure string `json:"failure,omitempty"`
+}
+
+// PreferenceDifferentialPricingRequest contains information about differential pricing configuration.
+type PreferenceDifferentialPricingRequest struct {
+	ID int64 `json:"id,omitempty"`
+}
+
+// PreferenceItemRequest represents a purchased item.
+type PreferenceItemRequest struct {
+	ID          string  `json:"id,omitempty"`
+	Title       string  `json:"title,omitempty"`
+	Description string  `json:"description,omitempty"`
+	PictureURL  string  `json:"picture_url,omitempty"`
+	CategoryID  string  `json:"category_id,omitempty"`
+	Quantity    int     `json:"quantity,omitempty"`
+	UnitPrice   float64 `json:"unit_price,omitempty"`
+	CurrencyID  string  `json:"currency_id,omitempty"`
+}
+
+// PreferencePayerRequest contains payer information in the preference.
+type PreferencePayerRequest struct {
+	Name           string                 `json:"name,omitempty"`
+	Surname        string                 `json:"surname,omitempty"`
+	Email          string                 `json:"email,omitempty"`
+	Phone          *PhoneRequest          `json:"phone,omitempty"`
+	Identification *IdentificationRequest `json:"identification,omitempty"`
+	Address        *AddressRequest        `json:"address,omitempty"`
+	DateCreated    *time.Time             `json:"date_created,omitempty"`
```

Comment:
Organizar por tipos e separar os tipos primitivos.
Remover os ponteiros
```suggestion
	Name           string                 `json:"name,omitempty"`
	Surname        string                 `json:"surname,omitempty"`
	Email          string                 `json:"email,omitempty"`
	
	DateCreated    *time.Time             `json:"date_created,omitempty"`
	Phone          PhoneRequest          `json:"phone,omitempty"`
	Identification IdentificationRequest `json:"identification,omitempty"`
	Address        AddressRequest        `json:"address,omitempty"`
```